lib/srv/desktop/rdp/rdpclient/src/lib.rs

```diff
@@ -247,12 +247,27 @@ fn connect_rdp_inner(
         KeyboardLayout::US,
         "rdp-rs",
     );
-    // Client for the "rdpdr" channel - smartcard emulation.
+
+    let tdp_sd_acknowledge = Box::new(move |ack: SharedDirectoryAcknowledge| -> RdpResult<()> {
```

Comment:
I've decided that we're moving away from this. 

It's possible to name this something else, and pass data as function parameters and/or differently named structs (and so on with [other similar functions I add in the future](https://github.com/gravitational/teleport/pull/12665/files#diff-983433889acb8e2c478d0a04b26bb477bd81271f29515f1ee6c56fa881e2880aR102-R105)), but ultimately that would just be a slight of hand. Functionally, this client would still be carrying out the same process of converting RDP into TDP and vice versa.

I can see an argument for treating this client as a "pure" RDP client, which blindly takes RDP messages it receives and passes them into Go, and which exposes an API that only allows users to call functions that directly send RDP messages to the Windows server. That would require some refactoring of our existing code, and a major overhaul of my big stack of existing PR's. Such a design would offer the benefit of pushing much of the Teleport specific functionality out of this client (besides the fact that we use Go). It would allow us to do things like, for example, swapping out TDP for another translation protocol (or just using RDP directly) without changing this client.

My thought is that we don't have a good reason to refactor this client in that way right now, because we don't have any plans for moving away from TDP. Granted I might be being biased due to my personal investment in building this system out without that design goal in mind, or I may be missing some other important line of reasoning.

Reply:
Hmm, the crate is literally called "rdpclient" so I'm of the mind it should only do RDP.

Reply:
Putting this here for documentation: we discussed on a call that our `rdpclient` primarily does RDP at the low level protocol. It will also have functionality to create data structures and call functions with those data structures that are named for the TDP messages they stand for, but it remains blind to the low level details of TDP. After this implementation is working, we should decide whether our naming conventions should change in order to make this separation more explicit.